includes/wikia/models/NavigationModel.class.php

```diff
@@ -222,6 +222,24 @@ function () use ( $type, $source, $maxChildrenAtLevel ) {
 		return $menuData;
 	}
 
+	public function getTreeFromText( $wikiText ) {
+		$this->setShouldTranslateContent( false );
+		$menuData = [];
+
+		$this->menuNodes = $this->parseText( $wikiText, [
+			$this->wg->maxLevelOneNavElements,
+			$this->wg->maxLevelTwoNavElements,
+			$this->wg->maxLevelThreeNavElements,
+		], true );
+
+		foreach ( $this->menuNodes[0][self::CHILDREN] as $id ) {
+			$menuData[] = $this->recursiveConvertMenuNodeToArray( $id );
+		}
+		$this->setShouldTranslateContent( true );
+
+		return $menuData;
+	}
+
 	/*
 	 * TODO we should refactor whole model when we remove Oasis
```

Comment:
@kvas-damian was such an optimist, back in the day (https://github.com/Wikia/app/commit/e431ee3c1d691a99dcd556cd248670aa55613a01#diff-f5c754c2b758d753c95dbffc34b8f45cR193) :smile: